repo/blob/azure/azure_storage_test.go

```diff
@@ -270,3 +273,175 @@ func TestAzureStorageInvalidCreds(t *testing.T) {
 		t.Errorf("unexpected success connecting to Azure blob storage, wanted error")
 	}
 }
+
+// TestAzureStorageRansomwareProtection runs through the behaviour of Azure ransomware protection.
+// 1. blob is created then the retention is extended.
+// 2. blob is logically deleted while the retention period is in place, by creating a delete marker (d_) file.
+// 3. delete marker blob is extended.
+// 4. original blob is deleted.
+// 5. delete marker blob further extension fails, then it is deleted.
+func TestAzureStorageRansomwareProtection(t *testing.T) {
```

Comment:
The feature is really immutability protection. _Ransomware_ is an application, but it is not the only one. It is also useful for recovering from (kopia) defects (aka bugs), or accidental deletions or corruptions.

```suggestion
func TestAzureStorageImmutabilityProtection(t *testing.T) {
```
Also, this test and the related helpers can be in a separate file, for example `azure_immutability_test.go`, that may also help with the collisions for the imported packages.